### PR TITLE
Forecasting: boilerplate only

### PR DESCRIPTION
**Pull request by romanlutz**
## Description

This takes the boilerplate parts out of #1874 and should make it a lot easier to review!

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/forecasting/src/lib/ForecastingDashboard/ForecastingDashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+} from "@responsible-ai/core-ui";
+import React from "react";
+
+export class ForecastingDashboard extends React.Component {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public render(): React.ReactNode {
+    return <Text>Placeholder</Text>;
```
Comment (gaugup) [inline libs/forecasting/src/lib/ForecastingDashboard/ForecastingDashboard.tsx:17]: Is `Placeholder` a string or a variable?

**Review by gaugup — APPROVED** [review]

File: libs/forecasting/src/lib/ForecastingDashboard/ForecastingDashboard.tsx (hunk 1 above)
Comment (romanlutz) [reply libs/forecasting/src/lib/ForecastingDashboard/ForecastingDashboard.tsx:17]: A string. And it's going to vanish in the next PR.

**Review by imatiach-msft — APPROVED** [review]

**Comment by imatiach-msft** [comment]
can you mention the command used to generate the files in the PR description?  Just for posterity.

**Comment by romanlutz** [comment]
> can you mention the command used to generate the files in the PR description?  Just for posterity.

I'd love to but this was done by John. I think it's an nx command. @xuke444 set it up do he might know.

**Review by vinuthakaranth — APPROVED** [review]

File: libs/forecasting/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  "extends": ["plugin:@nrwl/nx/react", "../../.eslintrc.json"],
+  "ignorePatterns": ["!**/*"],
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
+      "rules": {}
+    },
+    {
+      "files": ["*.ts", "*.tsx"],
+      "rules": {}
```
Comment (tongyu-microsoft) [inline libs/forecasting/.eslintrc.json:11]: can we add `"max-lines": "error"` in the rules for ts, tsx files
In the refactor work that I was working on, we are trying to limit file lines, so we reduce number of very large files

**Review by tongyu-microsoft — APPROVED** [review]

File: libs/forecasting/.eslintrc.json (hunk 2 above)
Comment (tongyu-microsoft) [reply libs/forecasting/.eslintrc.json:11]: this could be in next PR

Comment (romanlutz) [reply libs/forecasting/.eslintrc.json:11]: For new projects we should absolutely do that! It was actually here because it complained at some point when I went past 300 lines 😆 I subsequently broke up my files into smaller ones. Not sure where that rule went, let me check!

File: apps/dashboard/src/model-assessment-forecasting/App.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    LocalOrGlobalAndTestReq: [
+      { displayText: "LocalOrGlobalAndTestReq", format: "text" }
+    ],
+    PredictorReq: [{ displayText: "PredictorReq", format: "text" }],
+    TestReq: [{ displayText: "TestReq", format: "text" }]
+  };
+
+  public render(): React.ReactNode {
+    if (this.props.modelExplanationData) {
+      for (const exp of this.props.modelExplanationData) {
```
Comment (hawestra) [inline apps/dashboard/src/model-assessment-forecasting/App.tsx:34]: nit: this if is not needed right ? you could just iterate through this.props.modelExplanationData which if it's not defined wouldn't do anything

File: apps/dashboard/src/app/App.tsx [hunk 4]
```diff
@@ -175,6 +176,21 @@ export class App extends React.Component<IAppSetting, IAppState> {
...
+                this.state.dataset
+              ]}
+              theme={themes[this.state.theme]}
+              language={Language[this.state.language]}
+              version={
+                applications[this.state.application].versions[
+                  this.state.version
+                ]
+              }
+              featureFlights={parseFeatureFlights(this.state.featureFlights)}
```
Comment (hawestra) [inline apps/dashboard/src/app/App.tsx:191]: I'm guessing you copied this from the lines above but we're calling parseFeatureFlights a lot; could we save as a local variable and use that instead ? or is there any reason that this.state.featureFlights is not the parsed featureFlights ?

**Review by hawestra — APPROVED** [review]

File: apps/dashboard/src/app/App.tsx (hunk 4 above)
Comment (romanlutz) [reply apps/dashboard/src/app/App.tsx:191]: This is a complicated one 🙂 
If I remember correctly it would be easiest to represent it as a string[] everywhere, BUT this is one of the dropdowns in the test env so that we can try it out with or without flights (and also combinations thereof). That, in turn, means we need to be able to encode it in the URL, e.g., 
```
/modelAssessment/adultCensusIncomeData/1%20-%20light/en/1/flight1&flight2&flight3
```
So then at some point we have to parse it from this string to string[]. I think that part is intuitive, but you asked a great question: why not have it as string[] on the state? 
The answer is that the state consists of `IAppSetting`s and those defined as follows
```
export const routeKey = [
  "application",
  "dataset",
  "theme",
  "language",
  "version",
  "featureFlights"
] as const;
export type IAppSetting = {
  [key in typeof routeKey[number]]?: string;
};
```
In other words, we have no choice but to make it a string. We could, of course, have a duplicate and try to keep them in sync, but repeating the parse function a few times seems like a small price to pay in order not to have to keep them synced.

I may very well have missed something somewhere that makes this easily solvable. In that case, we should definitely adjust. Wdyt?

File: apps/dashboard/src/model-assessment-forecasting/App.tsx (hunk 3 above)
Comment (romanlutz) [reply apps/dashboard/src/model-assessment-forecasting/App.tsx:34]: This is indeed a strange way of writing this. Let me adjust that....

File: libs/forecasting/.eslintrc.json (hunk 2 above)
Comment (romanlutz) [reply libs/forecasting/.eslintrc.json:11]: Ah I think I get it now! This is on with "error" by default via .eslintrc\.eslintrc.custom.eslintrc, and some of the projects have it turned off, but not forecasting, so it is on.
